### ISSUE TITLE
does swap df.lon and df.lat matter? how does to_crs work?

### ISSUE THREAD

**Issue by shuai-zhou** [geopandas/geopandas#1530]
I remembered seeing somebody raised this question before, i.e., does it matter to swap `df.lon` and `df.lat` when creating GeoDataFrame from raw lat/long data? Also, I was confused about using `to_crs`. The following are the examples:

> %matplotlib inline
> import datetime
> import pandas as pd
> import geopandas as gpd
> import matplotlib.pyplot as plt

> dict = {'lat':[40.71,40.72,40.76,40.53,40.86,40.90,40.78,40.60], 'lon':[-73.92,-73.98,-73.90,-74.21,-72.77,-73.90,-73.78,-73.76]}
> df = pd.DataFrame(dict)

> gdf1 = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.lon, df.lat))
> gdf2 = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.lat, df.lon))

The following are the dataframe for gdf1 and gdf2, obviously, the elements in geometry columns are swapped. **Does this affect further spatial analyses? Are there some rules of thumb when creating GeoDataFrame from raw lat/lon data like this?**
![1](https://user-images.githubusercontent.com/22311821/88249162-94eaad80-cc71-11ea-84a8-55c3dfc41776.jpg)
![2](https://user-images.githubusercontent.com/22311821/88249163-95834400-cc71-11ea-92b9-3b4e0c8b0c63.jpg)

Another related issue is the use of `to_crs`. I will include `crs='epsg:3857'` in the building of GeoDataFrame, the geometry column of the dataframe should look like this (I am using the `df.lon, df.lat` approach) because it is what the geometry column should look like under `EPSG: 3857`:
![3](https://user-images.githubusercontent.com/22311821/88249611-f52e1f00-cc72-11ea-98a7-da5c964790c7.jpg)
However, when I included `crs='epsg:3857'`
> gdf3857 = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.lon, df.lat), crs='epsg:3857')
> gdf3857.crs

![4](https://user-images.githubusercontent.com/22311821/88249789-800f1980-cc73-11ea-8a18-3fa9e8a57cb5.jpg)
The `crs` information suggests that the GeoDataFrame is projected in `EPSG: 3857`. **However, the geometry column still shows the original lon/lat pairs. What is the point to include `crs='epsg:3857'` in the building of GeoDataFrame if it did not transform the geometry column? Or, what is the right way to build GeoDataFrame and project it simultaneously?**

**Comment by shuai-zhou** [comment]
I found even weirder thing. I was using Jupyter Notebook on a Windows 10 machine. See the examples:
> dict = {'lat':[40.71,40.72,40.76,40.53,40.86,40.90,40.78,40.60], 'lon':[-73.92,-73.98,-73.90,-74.21,-72.77,-73.90,-73.78,-73.76]}
> df = pd.DataFrame(dict)

> gdf1 = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.lon, df.lat))
> gdf1

![11](https://user-images.githubusercontent.com/22311821/88250733-7aff9980-cc76-11ea-8fd8-7bd14a4b496a.jpg)
> gdf2 = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.lat, df.lon))
> gdf2

![12](https://user-images.githubusercontent.com/22311821/88250766-97033b00-cc76-11ea-8a6a-4f96d3af195d.jpg)
**When I build the two GeoDataFrames in separate cells, the elements of the geometry columns are swapped!!**
> gdf3 = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.lon, df.lat))
> gdf4 = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.lat, df.lon))

![g3](https://user-images.githubusercontent.com/22311821/88250864-e21d4e00-cc76-11ea-80cd-ecc68936d6f5.jpg)
![g4](https://user-images.githubusercontent.com/22311821/88250865-e21d4e00-cc76-11ea-9a2a-572afa42b46d.jpg)
**When I build the two GeoDataFrames in the same cell, the geometry columns are the same even when `df.lon` and `df.lat` are swapped!!**

**Comment by martinfleis** [comment]
> does it matter to swap df.lon and df.lat when creating GeoDataFrame from raw lat/long data?

Of course, it does. If you swap coordinates, you are capturing different place on the Earth. (40, 5) is in east Africa, while (5, 40) in the Mediterranean sea. 

>  Are there some rules of thumb when creating GeoDataFrame from raw lat/lon data like this?

Longitude is `x`, latitude is `y` for the use in GeoPandas.


> Or, what is the right way to build GeoDataFrame and project it simultaneously?

You have to specify the CRS of coordinates you are passing in, and then you can reproject it to another one.

```py
gdf = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.lon, df.lat), crs='epsg:4326')
gdf3857 = gdf.to_crs(3857)
```

> When I build the two GeoDataFrames in the same cell, the geometry columns are the same even when df.lon and df.lat are swapped!!

That is strange, indeed. Not sure what causes it though.

**Comment by jorisvandenbossche** [comment]
(I think this is a topic we should try to better explain / highlight in the docs)

**Comment by shuai-zhou** [comment]
Thanks y'all for your help.
